letsencrypt/client/crypto_util.py

```diff
@@ -189,3 +190,43 @@ def get_cert_info(filename):
         "serial": cert.get_serial_number(),
         "pub_key": "RSA " + str(cert.get_pubkey().size() * 8),
     }
+
+
+def get_sans_from_csr(csr):
+    """Get list of Subject Alternative Names from signing request.
+
+    :param str csr: Certificate Signing Request in PEM format
+
+    :returns: List of referenced subject alternative names
+    :rtype: list
+    """
+    # TODO: This is a temporary solution involving parsing the .as_text()
+    #       output because there doesn't seem to be a built-in feature in
+    #       any Python cryptography module that performs this function.
+    #       In the future we should try to replace this with a more direct
+    #       use of relevant OpenSSL or other X509-parsing APIs.
+    req = M2Crypto.X509.load_request_string(csr)
+    text = req.as_text().split("\n")
+    if len(text) < 2 or text[0] != "Certificate Request:" or \
+                        text[1] != "    Data:":
+        raise ValueError("Unable to parse CSR")
+    text = text[2:]
+    while text and text[0] != "        Attributes:":
```

Comment:
This seems a bit fragile to me.  Different versions could have slightly different spacing/structure.  I think I would be in favor of a slightly more powerful regex taking into account just names that are unlikely to change.

I do think this feature is vital though as we will need to offer the ability to input only a CSR (no associated private key required) for launch.  When we integrate it we can also try to fail gracefully if this function fails to return valid names.